README.md

```diff
@@ -37,6 +40,50 @@ int main()
 ```
 
 `
+# More advanced usage
+You can forget about the order of parameters in your code
+
+``` c++
+#include <boxed-cpp/boxed.hpp>
+
+// Create unique structures
+namespace tags { struct Speed{}; struct Permittivity{}; struct Permeability{}; }
+
+using Speed = boxed::boxed<double,tags::Speed>;
+using Permittivity = boxed::boxed<double,tags::Permittivity>;
+using Permeability = boxed::boxed<double,tags::Permeability>;
+
+Speed wave_speed_inside(Permittivity epsilon, Permeability mu)
+{
+    return Speed(1.0 / std::sqrt(unbox(epsilon) * unbox(mu)));
+};
+
+template <typename T, typename S>
+Speed wave_speed(T t, S s) // recursive variadic function
+{
+    if constexpr (std::is_same_v<T, Permittivity>)
+    {
+        return wave_speed_inside(t, s);
+    }
+    else
+    {
+        return wave_speed_inside(s, t);
+    }
+}
+
+
+int main()
+{
+    auto vacuum_permittivity = Permittivity(8.85418781762039e-12);
+    auto pi = 3.14159265358979323846;
+    auto vacuum_permeability = Permeability(4 * pi * 1e-7);
```

Comment:
for the fun of it, all 3 of them could be made `constexpr`, right? ;-)

Reply:
> 

Done :)